arch/armv7a/memcpy.S

```diff
@@ -0,0 +1,361 @@
+#define DST r0
+#define SRC r1
+#define LEN r8
+#define DST_RET r12
+
+.arm
+.syntax unified
+
+.text
+.align 4
+
+/* void *memcpy(void *dst, const void *src, size_t l) */
+.globl memcpy
+.type memcpy, %function
+memcpy:
+	str r8, [sp, #-8]!
+	mov r8, r2
+	cmp LEN, #64
+	mov DST_RET, DST /* preserve return value */
+
+	bhs .LblkCopy
+
+	/* less than 64 bytes - always copy as if block was always unaligned */
+
+.Ltail63Unaligned:
+	/* unaligned copy, 0-63 bytes */
+
+	/* r3 = LEN / 4 */
+	movs r3, LEN, lsr #2
+	beq .Ltail63Un0
+
+.Ltail63Un4:
+	ldr r2, [SRC], #4
+	str r2, [DST], #4
+	subs r3, #1
+	bne .Ltail63Un4
```

Comment:
Is unaligned memory access allowed?

Reply:
yes, `ldr` and `str` instructions allow for unaligned addresses (but with some performance penalty - which doesn't really matter here as we're copying only up to 60 bytes that way). [ARM documentation](https://developer.arm.com/documentation/den0013/d/Optimizing-Code-to-Run-on-ARM-Processors/Source-code-modifications/Unaligned-access)

Reply:
I meant your project, not in general.

Reply:
After some discussion I think it will be beneficial to allow it at least in this case. It will need small CPU initialization change, though. @nalajcie What do you think? Enabling unaligned access will make memcpy simpler and faster (as unaligned 4 byte access should still be faster than 4 separate 1 byte accesses)

Reply:
Probably some invalid info I've found on internet (https://s-o-c.org/does-arm-allow-unaligned-access/). The reference page I've linked in previous comment indeed says that U=1 on all modern ARMs so `STR/LDR` should just produce unaligned access - please just scratch out that part of the comment :).

Sorry for providing invalid info. If we choose to keep the unaligned access, maybe we should explicitly say it in comment to ensure nobody would think in the future that it's an error on our part?